AWSS3/AWSS3TransferUtility.m

```diff
@@ -1801,78 +1801,86 @@ - (void)enumerateToAssignBlocksForUploadTask:(void (^)(AWSS3TransferUtilityUploa
-                
-                downloadBlocksAssigner(transferUtilityDownloadTask, &progressBlock, &completionHandler);
-                
-                if (progressBlock) {
-                    transferUtilityDownloadTask.expression.progressBlock = progressBlock;
-                }
-                if (completionHandler) {
-                    transferUtilityDownloadTask.expression.completionHandler = completionHandler;
-                }
-            }
-        }
-    }
-}
-
--(void)enumerateToAssignBlocksForUploadTask:(void (^)(AWSS3TransferUtilityUploadTask *uploadTask,
-                                                      AWSS3TransferUtilityProgressBlock *uploadProgressBlockReference,
-                                                      AWSS3TransferUtilityUploadCompletionHandlerBlock *completionHandlerReference))uploadBlocksAssigner
-              multiPartUploadBlocksAssigner: (void (^) (AWSS3TransferUtilityMultiPartUploadTask *multiPartUploadTask,
-                                                        AWSS3TransferUtilityMultiPartProgressBlock *multiPartUploadProgressBlockReference,
-                                                        AWSS3TransferUtilityMultiPartUploadCompletionHandlerBlock *completionHandlerReference)) multiPartUploadBlocksAssigner
-                     downloadBlocksAssigner:(void (^)(AWSS3TransferUtilityDownloadTask *downloadTask,
-                                                      AWSS3TransferUtilityProgressBlock *downloadProgressBlockReference,
-                                                      AWSS3TransferUtilityDownloadCompletionHandlerBlock *completionHandlerReference))downloadBlocksAssigner {
-   
-    [self enumerateToAssignBlocksForUploadTask:uploadBlocksAssigner  downloadTask:downloadBlocksAssigner];
-   
-    // Iterate through MultiPartUploadTasks
-    for (id key in [self.taskDictionary allKeys]) {
-        id value = [self.taskDictionary objectForKey:key];
-        if ([value isKindOfClass:[AWSS3TransferUtilityMultiPartUploadTask class]]) {
+        } else if (multiPartUploadBlocksAssigner && [value isKindOfClass:[AWSS3TransferUtilityMultiPartUploadTask class]]) {
             AWSS3TransferUtilityMultiPartUploadTask *task = value;
             if (multiPartUploadBlocksAssigner) {
                 AWSS3TransferUtilityMultiPartProgressBlock progressBlock = nil;
                 AWSS3TransferUtilityMultiPartUploadCompletionHandlerBlock completionHandler = nil;
+
                 multiPartUploadBlocksAssigner(task, &progressBlock, &completionHandler);
+
                 if (progressBlock) {
                     task.expression.progressBlock = progressBlock;
                 }
                 if (completionHandler) {
                     task.expression.completionHandler = completionHandler;
                 }
             }
+        } else if ([value isKindOfClass:[AWSS3TransferUtilityDownloadTask class]]) {
```

Comment:
Add an extra check for `downloadBlocksAssigner` here like above and remove the if statement from below?